src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1422,4 +1422,13 @@
   <data name="DoNotUseOutAttributeStringPInvokeParametersTitle" xml:space="preserve">
     <value>Do not use 'OutAttribute' on string parameters for P/Invokes</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="AvoidAssemblyLocationInSingleFileTitle" xml:space="preserve">
+    <value>Avoid using accessing Assembly file path when publishing as a single-file</value>
+  </data>
+  <data name="AvoidAssemblyLocationInSingleFileMessage" xml:space="preserve">
+    <value>'{0}' always returns an empty string for assemblies embedded in a single-file app. If the path to the app directory is needed, consider calling 'System.AppContext.BaseDirectory'.</value>
+  </data>
+  <data name="AvoidAssemblyGetFilesInSingleFile" xml:space="preserve">
+    <value>Assemblies embedded in a single-file app cannot have additional files in the manifest.</value>
```

Comment:
Maybe it would be better to similarly to Location describe the actual behavior. In this case the API is going to throw an exception (it already does for in-memory assemblies outside of single-file).

Reply:
The issue at https://github.com/dotnet/runtime/issues/38405 seems to imply the expected behavior is to return null or empty array.

Reply:
We'll probably have to update that issue. More recent discussions:
* `Assembly.CodeBase`: https://github.com/dotnet/runtime/issues/40087 - the agreement is to actually throw `PlatformNotSupported` - it's "safer" and we're obsoleting these anyway, so making them work feels contraproductive.
* Module.FullyQualifiedName https://github.com/dotnet/runtime/issues/40103 - should return `<Unknown>`. This will make it work the same as memory-loaded assemblies. This also means that `GetFile`/`GetFiles` will throw (as it does in non-single-file for memory-loaded assemblies).

Reply:
What about `AssemblyName.CodeBase`? AFAIK that's not even obsolete

Reply:
I personally would expect `Assembly.CodeBase` and `AssemblyName.CodeBase` to behave the same. There has been discussion on it, but it sort of disappeared: https://github.com/dotnet/runtime/issues/31127
I'll start it again.